File: python/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maximum_profit = 0
        l = 0

        for r in range(len(prices)):
            profit = prices[r] - prices[l]
            maximum_profit = max(profit, maximum_profit)
            if prices[l] > prices[r]:
                l = r
        return maximum_profit

File: javascript/easy/best-time-to-buy-and-sell-stock.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let maxPrice = 0;
        let l = 0;

        for(let r = 1; r < prices.length; r++){
            let profit = prices[r] - prices[l];
            maxPrice = Math.max(maxPrice, profit);
            if(prices[r] < prices[l]) l = r
        }
        return maxPrice;
    }
}
